### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by joelhawksley**
This PR:

1) Updates our docs for grammar and clarity.
2) Updates our PR template, taking inspiration from the `github` template.
3) Updates the Helpers docs based on notes from @kevin-j-m's RailsConf talk.
4) Adds notes to the changelog about an open role at GitHub and our RailsConf happy hour ❤️

### CODE REVIEW

File: docs/guide/helpers.md [hunk 1]
```diff
@@ -6,7 +6,19 @@ parent: Guide
 
 # Helpers
 
-Helper methods can be used through the `helpers` proxy:
+Helpers must be included in order to be used:
+
+```ruby
+class UserComponent < ViewComponent::Base
+  include IconHelper
```
Comment (kevin-j-m) [inline docs/guide/helpers.md:13]: With this order flipping, you may want to define/introduce the `IconHelper` module in this section (as opposed to below in the `helpers` proxy recommendation).

Comment (joelhawksley) [reply docs/guide/helpers.md:19]: Oh, yes! Thanks for catching this ❤️

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,14 @@ title: Changelog
 
     *Joel Hawksley*
 
+* Clean up grammar in documentation.
+
+    *Joel Hawksley*
+
+* The ViewComponent team at GitHub is hiring! We're looking for a Rails engineer with accessibility experience: [https://boards.greenhouse.io/github/jobs/3802139]. Reach out to joelhawksley@github.com with any questions!
```
Comment (Spone) [inline docs/CHANGELOG.md:24]: @joelhawksley The link seems broken.

File: docs/guide/helpers.md [hunk 3]
```diff
@@ -6,39 +6,39 @@ parent: Guide
 
 # Helpers
 
-Helper methods can be used through the `helpers` proxy:
+Helpers must be included in order to be used:
```
Comment (Spone) [inline docs/guide/helpers.md:9]:
```suggestion
Helpers must be included to be used:
```

File: docs/logo.md [hunk 4]
```diff
@@ -52,12 +52,12 @@ Logomark for dark backgrounds ([SVG](logo/viewcomponent-white-logomark.svg) •
 
 ## Usage
 
-You may use these marks to refer to ViewComponent in a way where it‘s clear that you‘re simply referring to the project, not claiming endorsement or association.
+Use these marks to refer to the ViewComponent project in a way that does not claim endorsement or association.
```
Comment (Spone) [inline docs/logo.md:55]:
```suggestion
Use these marks to refer to the ViewComponent project in a way that doesn't claim endorsement or association.
```

File: docs/CHANGELOG.md (hunk 2 above)
Comment (dylanatsmith) [reply docs/CHANGELOG.md]: Looks like this is the active URL: https://boards.greenhouse.io/github/jobs/4020166

**Review by Spone — APPROVED** [review]

**Pull request merged**